packages/database/src/core/snap/IndexMap.ts

```diff
@@ -75,7 +75,7 @@ export class IndexMap {
       // regular child map
       return null;
     } else {
-      return sortedMap;
+      return sortedMap as SortedMap<NamedNode, Node>;
```

Comment:
You made SafeGet type-safe below. I think you should need to update the index.d.ts of the util package as this `any` would go away.

Reply:
This isn't about that though. The "fallback type" here is object, which is a subtype of `SortedMap`, so TS can't narrow the type to `SortedMap`.

I reversed the condition and used `instanceof` instead.